src/os.cc

```diff
@@ -353,6 +354,20 @@ buffered_file file::fdopen(const char* mode) {
   return bf;
 }
 
+#  if defined(_WIN32) && !defined(__MINGW32__)
+file file::open_windows_file(wcstring_view path, int oflag) {
+  int fd_ = -1;
+  using mode_t = int;
+  constexpr mode_t mode =
+      S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH;
```

Comment:
Let's move to a constant, e.g. `default_open_mode` and reuse here and above to make sure modes are consistent.